frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1427,315 +1353,60 @@ function QueryBuilder<T extends SearchParamsInput | ErrorSearchParamsInput>(
-									value,
-							},
-						}
-					case 'exists':
-						return { exists: { field: name } }
-					case 'between_date':
-						return {
-							range: {
-								[name]: {
-									gte: getAbsoluteStartTime(value),
-									lte: getAbsoluteEndTime(value),
-								},
-							},
-						}
-					case 'between_time':
-						return {
-							range: {
-								[name]: {
-									gte:
-										Number(value?.split('_')[0]) *
-										60 *
-										1000,
-									...(Number(value?.split('_')[1]) ===
-									TIME_MAX_LENGTH
-										? null
-										: {
-												lte:
-													Number(
-														value?.split('_')[1],
-													) *
-													60 *
-													1000,
-										  }),
-								},
-							},
-						}
-					case 'between':
-						return {
-							range: {
-								[name]: {
-									gte: Number(value?.split('_')[0]),
-									...(Number(value?.split('_')[1]) ===
-									RANGE_MAX_LENGTH
-										? null
-										: {
-												lte: Number(
-													value?.split('_')[1],
-												),
-										  }),
-								},
-							},
-						}
-				}
-			} else {
-				const key = field.value
-				switch (op) {
-					case 'is':
-						return {
-							term: { 'fields.KeyValue': `${key}_${value}` },
-						}
-					case 'contains':
-						return {
-							wildcard: {
-								'fields.KeyValue': `${key}_*${value}*`,
-							},
-						}
-					case 'matches':
-						return {
-							regexp: {
-								'fields.KeyValue': `${key}_${value}`,
-							},
-						}
-					case 'exists':
-						return { term: { 'fields.Key': key } }
-				}
-			}
-		},
-		[getCustomFieldOptions, admin],
-	)
-
-	const parseRuleImpl = useCallback(
-		(
-			field: SelectOption,
-			op: Operator,
-			multiValue: MultiselectOption,
-		): any => {
-			if (isNegative(op)) {
-				return {
-					bool: {
-						must_not: {
-							...parseRuleImpl(
-								field,
-								NEGATION_MAP[op],
-								multiValue,
-							),
-						},
-					},
-				}
-			} else if (hasArguments(op)) {
-				return {
-					bool: {
-						should: multiValue.options.map(({ value }) =>
-							parseInner(field, op, value),
-						),
-					},
-				}
-			} else {
-				return parseInner(field, op)
-			}
-		},
-		[parseInner],
-	)
-
-	const parseRule = useCallback(
-		(rule: RuleProps): any => {
-			const field = rule.field!
-			const multiValue = rule.val!
-			const op = rule.op!
-
-			return parseRuleImpl(field, op, multiValue)
-		},
-		[parseRuleImpl],
-	)
 	const { data: appVersionData } = useGetAppVersionsQuery({
 		variables: { project_id: projectId! },
 		skip: !projectId,
 	})
 
 	const [currentRule, setCurrentRule] = useState<RuleProps | undefined>()
+
+	const {
+		isAnd: serializedIsAnd,
+		rules: serializedRules,
+	}: { isAnd: boolean; rules: any } = JSON.parse(searchQuery)
+	const startingRules = deserializeRules(serializedRules)
+	const [isAnd, toggleIsAnd] = useToggle(serializedIsAnd)
+	const [rules, setRules] = useState<RuleProps[]>(startingRules)
```

Comment:
set the default rules / date from the search context instead of waiting on a use effect / second render